### ISSUE TITLE
Intersects is not consistent for 2 Rectangles

### ISSUE THREAD

**Issue by peterbraden** [georust/geo#419]
Butchered example here:

https://play.rust-lang.org/?version=stable&mode=debug&edition=2018&gist=2e89595e108b86c756ee05d9516ad7dc

If we have 2 Rect<f64>'s:
```
    let a = Rect { min: Coordinate { x: 9.228515625, y: 46.83013364044739 }, max: Coordinate { x: 9.2724609375, y: 46.86019101567026 } };
    let b = Rect { min: Coordinate { x: 9.17953, y: 46.82018 }, max: Coordinate { x: 9.26309, y: 46.88099 } };
```
then a.intersects(b) should equal b.intersects(a).

Further illustrated here:
https://play.rust-lang.org/?version=stable&mode=debug&edition=2018&gist=a0c732369e3e2852fb2cfde7b98bb179

Neither contain one another.

Or am I misunderstanding the algorithm?

**Comment by urschrei** [comment]
I can reproduce this in `geo` – looks like a bug to me.

**Comment by urschrei** [comment]
I've instrumented the function, and all the intermediate results are correct  
(someone please double-check the below, though):

| LHS       | Predicate | RHS  | Result  |
|-------------------|----|----------|-------|
 9.228515625       | >= | 9.17953  | true  |
| 9.228515625       | <= | 9.26309  | true  |
| 9.2724609375      | >= | 9.17953  | true  |
| 9.2724609375      | <= | 9.26309  | false |
| 46.83013364044739 | >= | 46.82018 | true  |
| 46.83013364044739 | <= | 46.88099 | true  |
| 46.86019101567026 | >= | 46.82018 | true  |
| 46.86019101567026 | <= | 46.88099 | true  |


| LHS      | Predicate | RHS               | Result |
|----------|-----------|-------------------|--------|
| 9.17953  | >=        | 9.228515625       | false  |
| 9.17953  | <=        | 9.2724609375      | true   |
| 9.26309  | >=        | 9.228515625       | true   |
| 9.26309  | <=        | 9.2724609375      | true   |
| 46.82018 | >=        | 46.83013364044739 | false  |
| 46.82018 | <=        | 46.86019101567026 | true   |
| 46.88099 | >=        | 46.83013364044739 | true   |
| 46.88099 | <=        | 46.86019101567026 | false  |


But the algorithm evaluates the results like so:

(A && B || C && D) && (E && F || G && H) _or_

`(true && true || true && false) && (true && true || true && true)` for `a.intersects(&b)`
Result: **true**

`(false && true || true && true) && (false && true || true && false)` for `b.intersects(&a)`
Result: **false**

Both of which are [the _correct_ answers](https://play.rust-lang.org/?version=stable&mode=debug&edition=2018&gist=c6887a2eea78e170fa3e3b14b7610107) for those predicate combinations. So do we have a simple logic error in the algorithm somewhere?

**Comment by urschrei** [comment]
@peterbraden Thanks for catching this – it looks like we had an incorrect test assertion failing to catch the bug in the existing implementation.

@frewsxcv We should probably cut a new release (and a `geo-types` release, since we've got a PR waiting for review) with this fix once we resolve #418…
